types/react-bootstrap-table/index.d.ts

```diff
@@ -12,48 +12,115 @@
 import { ComponentClass, Props, ReactElement } from 'react';
 import { EventEmitter } from 'events';
 
+/**
+ * Interface spec for sepcifying functionality to handle remotely
+ * 
+ * Consult [documentation](https://allenfang.github.io/react-bootstrap-table/docs.html#remote)
+ * for more info
+ * 
+ * @interface RemoteObjSpec
+ */
+export interface RemoteObjSpec {
+	/**
+	 * If set, cell edits will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	cellEdit?: boolean;
+	/**
+	 * If set insertions will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	insertRow?: boolean;
+	/**
+	 * If set deletion will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	dropRow?: boolean;
+	/**
+	 * If set filters will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	filter?: boolean;
+	/**
+	 * If set search will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	search?: boolean;
+	/**
+	 * If set, exporting CSV will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	exportCSV?: boolean;
+	/**
+	 * If set sorting will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	sort?: boolean;
+	/**
+	 * If set pagination will be handled remotely
+	 * 
+	 * @type {boolean}
+	 * @memberof RemoteObjSpec
+	 */
+	pagination?: boolean;
+}
+
 export interface BootstrapTableProps extends Props<BootstrapTable> {
 	/**
 	Use data to specify the data that you want to display on table.
 	*/
-    data: any[];
+	data: any[];
 	/**
 	If set, data is remote (use also fetchInfo)
 	*/
-    remote?: boolean,
+	remote?: (remobeObj: RemoteObjSpec) => RemoteObjSpec | boolean,	// Updated to support ^3.0.0
```

Comment:
this line is wrong, as the compiler is thinking it's just a function that must return `RemoteObjSpec | boolean`

it should be `boolean | (remobeObj: RemoteObjSpec) => RemoteObjSpec`

or put some parenthesis around

Reply:
Thanks for finding @pocesar. I've updated with the fix and submitted a new PR.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19695